### PR TITLE
build osg image

### PR DESCRIPTION
**Pull request by zhenz**

### CODE REVIEW

File: elements/osg/install.d/01-docker [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+if [ ${DIB_DEBUG_TRACE:-0} -gt 0 ]; then
+    set -x
+fi
+set -eu
+set -o pipefail
+
+# set up the repository
+sudo yum install -y yum-utils
```
Comment (diurnalist) [inline elements/osg/install.d/01-docker]: with CentOS 8 this is `dnf`, do we want to start using this or should this retain CentOS 7 compatibility?

**Review by diurnalist — APPROVED** [review]
This looks very solid! Nice work.

**Comment by diurnalist** [comment]
One thing that we may consider moving forward is making these "special" images have their own repo and use the base chameleon stuff as a separate element. I think this should be possible by cloning e.g. the centos chameleon element as a sibling directory and setting up the right build paths.

See https://github.com/ChameleonCloud/chameleon-ipa-builder/blob/master/dib/chameleon-ipa-hardware-manager/element-deps for element-deps example and https://github.com/ChameleonCloud/chameleon-ipa-builder/blob/master/tox.ini#L17 for how the element paths can be specified.

Something to think about for the future! I think it could also enable making some of the "special" stuff more generic. We could have them support Ubuntu and CentOS and then derive from a base element that does the same.

File: elements/osg/install.d/01-docker (hunk 1 above)
Comment (zhenz) [reply elements/osg/install.d/01-docker]: This builder is supposed to work for both centos7 and 8. I could do a check and use yum for centos 7 and dnf for centos8.

**Comment by zhenz** [comment]
> One thing that we may consider moving forward is making these "special" images have their own repo and use the base chameleon stuff as a separate element. I think this should be possible by cloning e.g. the centos chameleon element as a sibling directory and setting up the right build paths.
> 
> See https://github.com/ChameleonCloud/chameleon-ipa-builder/blob/master/dib/chameleon-ipa-hardware-manager/element-deps for element-deps example and https://github.com/ChameleonCloud/chameleon-ipa-builder/blob/master/tox.ini#L17 for how the element paths can be specified.
> 
> Something to think about for the future! I think it could also enable making some of the "special" stuff more generic. We could have them support Ubuntu and CentOS and then derive from a base element that does the same.

Thank you, @diurnalist ! This sounds good! I'll merge this as it is now to get the OSG stuff moving along and come back to this afterward!

**Pull request merged**